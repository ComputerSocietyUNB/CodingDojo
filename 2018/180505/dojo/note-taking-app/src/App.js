import React, { Component } from 'react'
import Note from './Note'
import ShowNotes from './ShowNotes'

class App extends Component {
  state = {
    value: '',
    listNotes: ['potato', 'oioi']
  }

  handleChange = (event) => {
    this.setState({
      value: event.target.value,
    })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <Note handleChange={this.handleChange} value={this.state.value} />
        <ShowNotes listNotes={this.state.listNotes} />
      </div>
    )
  }
}

export default App
