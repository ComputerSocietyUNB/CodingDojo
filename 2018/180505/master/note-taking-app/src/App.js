import React, { Component } from 'react'
import Note from './Note'
import NoteList from './NoteList'

class App extends Component {
  state = {
    noteList: [
      'my first note',
    ],
    current: 0,
  }

  getCurrentNote = () => {
    const { noteList, current } = this.state

    return noteList[current]
  }

  handleChange = (event) => {
    const { noteList, current } = this.state

    this.setState({
      noteList: [...noteList.slice(0, current), event.target.value, ...noteList.slice(current + 1)]
    })
  }

  removeNotes = (notesToBeRemoved) => {
    this.setState(({ noteList, current }) => ({
      noteList: noteList.filter((note, index) => !notesToBeRemoved.includes(index)),
      current: 0,
    }))
  }

  addNote = () => {
    this.setState(({ noteList }) => ({
      noteList: [...noteList, noteList.length],
      current: noteList.length,
    }))
  }

  changeSelection = (index) => {
    this.setState({
      current: index,
    })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <div className="App-intro">
          <NoteList
            list={this.state.noteList}
            addNote={this.addNote}
            removeNotes={this.removeNotes}
            changeSelection={this.changeSelection}
            current={this.state.current}
          />
          <Note
            note={this.getCurrentNote()}
            onNoteEdit={this.handleChange}
          />
        </div>
      </div>
    )
  }
}

export default App
