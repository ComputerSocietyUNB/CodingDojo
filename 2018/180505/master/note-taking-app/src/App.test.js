import React from 'react'
import { render, getByText } from 'react-testing-library'
import App from './App'

it('renders without crashing', () => {
  const { container, getByText } = render(<App />)
  expect(getByText('Welcome to React')).toMatchSnapshot()
})
