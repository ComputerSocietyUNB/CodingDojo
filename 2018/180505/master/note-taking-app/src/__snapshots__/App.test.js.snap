// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App edits notes 1`] = `
<ul
  class="list"
>
  <li
    class="note current"
  >
    <input
      type="checkbox"
      alt="remove note #1"
    />
    <span
      class="content"
    >
      My new value
    </span>
  </li>
</ul>
`;

exports[`App edits notes 2`] = `
<input
  type="text"
  placeholder="Your note goes here"
  value="My new value"
/>
`;

exports[`App renders without crashing 1`] = `
<h1
  class="App-title"
>
  Welcome to React
</h1>
`;

exports[`App types into the input 1`] = `
<input
  type="text"
  placeholder="Your note goes here"
  value="potato"
/>
`;
